app/mempool/senders_with_nonce.go

```diff
@@ -0,0 +1,63 @@
+package mempool
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+)
+
+func GetSendersWithNonce(tx sdk.Tx) ([]SenderWithNonce, error) {
+	if txWithExtensions, ok := tx.(authante.HasExtensionOptionsTx); ok {
+		opts := txWithExtensions.GetExtensionOptions()
+		if len(opts) > 0 && opts[0].GetTypeUrl() == "/ethermint.evm.v1.ExtensionOptionsEthereumTx" {
```

Comment:
NIT: Not possible to get the URL from the type directly? Or maybe use a constant